lib/WorseReflection/Bridge/PHPStan/DocblockParser/PHPStanDocblockParserFactory.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+namespace Phpactor\WorseReflection\Bridge\PHPStan\DocblockParser;
+
+use PHPStan\PhpDocParser\Parser\ParserException;
+use Phpactor\WorseReflection\Core\DocBlock\PlainDocblock;
+use Phpactor\WorseReflection\Core\DocBlock\DocBlock;
+use Phpactor\WorseReflection\Core\DocBlock\DocBlockFactory;
+use Phpactor\WorseReflection\Core\Reflection\ReflectionScope;
+use Phpactor\WorseReflection\Reflector;
+use PHPStan\PhpDocParser\Lexer\Lexer;
+use PHPStan\PhpDocParser\Parser\ConstExprParser;
+use PHPStan\PhpDocParser\Parser\PhpDocParser;
+use PHPStan\PhpDocParser\Parser\TokenIterator;
+use PHPStan\PhpDocParser\Parser\TypeParser;
+
+class PHPStanDocblockParserFactory implements DocBlockFactory
+{
+    private const SUPPORTED_TAGS = [
+        'assert',
+        'deprecated',
+        'extends',
+        'implements',
+        'method',
+        'mixin',
+        'param',
+        'property',
+        'return',
+        'template',
+        'template-covariant',
+        'template-extends',
+        'throws',
+        'type',
+        'var',
+    ];
+
+    private PhpDocParser $parser;
+
+    public function __construct(
+        private Reflector $reflector,
+        private Lexer $lexer = new Lexer(),
+        ?PhpDocParser $parser = null,
+    ) {
+        $parser ??= new PhpDocParser(new TypeParser(new ConstExprParser()), new ConstExprParser());
+        $this->parser = $parser;
+    }
+
+    public function create(string $docblock, ReflectionScope $scope): DocBlock
+    {
+        if (trim($docblock) === '') {
+            return new PlainDocblock();
+        }
+
+        // if no supported tags in the docblock, do not parse it
+        if (0 === preg_match(
+            sprintf('{@((psalm|phpstan|phan)-)?(%s)}', implode('|', self::SUPPORTED_TAGS)),
+            $docblock,
+            $matches
+        )) {
+            return new PlainDocblock($docblock);
```

Comment:
we should extract this "filter" to e.g. `Phpactor\WorseReflection\Core\DocBloc` so that it's used by both implementations (assuming we will run them side-by-side via. a strategy initially)

Reply:
Good :+1: . I was thinking about the same, but didn't want to introduce that prematurely

Reply:
I've settled with decorator because I did not fully understand how could I handle this with just static method, but if you have any suggestion, please do.